### DISCUSSION TITLE
How to refer to pivot entity when joining with the query builder?

### DISCUSSION THREAD

**Discussion by squivix** [mikro-orm/mikro-orm#4547] [Q&A] [answered]
I need to join a M:N relationship but also filter it based on some columns in the pivot entity. How can I do that in the query builder without changing the entity definition?

Something like:
```ts
const queryBuilder = em.createQueryBuilder(EntityA, "ea");
queryBuilder.select("*")
    .leftJoinAndSelect("ea.entityBs", "ea__entity_bs",{ "pivot_entity_a_entity_b": {joinedColumn: 1} })
```

Or alternatively if I can't refer to the pivot entity in the join conditions can I join by the pivot entity manually instead of specifying the field?

**Comment by B4nan** [comment, marked as answer]
You need to define 1:m relation to the pivot entity and use that.

**Reply by squivix** [reply]
I was hoping to avoid changing the entity definition for one query :/

But ok.

**Reply by B4nan** [reply]
Well, you can't map to properties that are not part of your entity definition, but you can always use a raw query, no shame in that.
